### PR TITLE
Cross-version e2e kube-proxy restart

### PR DESCRIPTION
**Pull request by thockin**
Partial fix for #19729

### CODE REVIEW

File: test/e2e/util.go [hunk 1]
```diff
@@ -2041,6 +2042,25 @@ func restartKubeProxy(host string) error {
 	if !providerIs("gce", "gke", "aws") {
 		return fmt.Errorf("unsupported provider: %s", testContext.Provider)
 	}
+	err1 := restartKubeProxyByInitD(host)
```
Comment (ihmccreery) [inline test/e2e/util.go]: Can we gate this on serverVersion instead?  See:

https://github.com/kubernetes/kubernetes/commit/a5d895ac49d5047f350f991de65076999b7e9d8f#diff-8349006db2c242fd7424e1dfb3295840R959

**Comment by ihmccreery** [comment]
Thanks @thockin.  If you think this isn't too ugly, I'd like to backport it to v1.0 too, to make those cross-version tests work.

**Comment by thockin** [comment]
That will check the API server version, which may or may not be the same as the node version?  Is that semantically correct to use ServerVersion here?

**Comment by thockin** [comment]
All the e2e failures here are unrelated - what is desired course of action?

**Comment by ihmccreery** [comment]
> That will check the API server version, which may or may not be the same as the node version? Is that semantically correct to use ServerVersion here?

Oh, probably not if this is a node-version issue?

> All the e2e failures here are unrelated - what is desired course of action?

Use `test this again issue: #<issue-number>`

**Comment by ihmccreery** [comment]
Also fixes #17965, as far as I can tell.

**Comment by ihmccreery** [comment]
@k8s-bot unit test this again please #20498

**Comment by ihmccreery** [comment]
@k8s-bot unit test this again please, issue #20498

**Comment by ihmccreery** [comment]
Re-adding LGTM; I used the wrong format for re-test.

**Pull request merged**